docs: describe how to manage embedded plugin registry for VS Code editor

max-cx opened the pull request:
<!-- 
Please use one of the following prefixes for the title:
docs: Documentation not including procedures. Engineering review is mandatory.
procedures: Documentation including procedures. Testing procedures is mandatory. Engineering and QE review is mandatory (Engineering can review on behalf of QE). 
chore: Routine, release, tooling, version upgrades.
fix: Fix build, language, links, or metadata.
-->

<!-- Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/main/CONTRIBUTING.adoc) before submitting a PR. -->

## What does this pull request change?
Changes from https://github.com/eclipse-che/che-docs/pull/2503 to `main`.

Adds instructions how to manage extensions for VS Code editor

## What issues does this pull request fix or reference?
https://issues.redhat.com/browse/CRW-3472

## Specify the version of the product this pull request applies to
`main`
cherry-pick to `7.56.x` (DS 3.3)
cherry-pick to `7.58.x` (DS 3.4)

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/main/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

max-cx committed:
outline and files ready for the SME

max-cx committed:
Update modules/administration-guide/pages/managing-ide-extensions.adoc

max-cx committed:
Update modules/administration-guide/pages/managing-ide-extensions.adoc

max-cx committed:
Update modules/administration-guide/pages/managing-ide-extensions.adoc

svor committed:
docs: describe how to manage embedded plugin registry for VS Code editor (#2503)

Signed-off-by: Valerii Svydenko <vsvydenk@redhat.com>
Co-authored-by: Nick Boldt <nboldt@redhat.com>
Co-authored-by: Max Leonov <mleonov@redhat.com>
Co-authored-by: Mario Loriedo <mario.loriedo@gmail.com>

nickboldt commented on modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,87 @@
+[id="adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance"]
+= Adding or removing pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance
+
+You can add or remove your choice of pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance that is deployed by {prod-short} to support air-gapped, offline, and proxy-restricted environments.
```
this could be simpler...

```suggestion
You can add or remove extensions in the embedded Open VSX registry instance deployed by {prod-short} to support offline and proxied environments.
```

nickboldt commented on modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,87 @@
+====
+* If the extension publisher is unavailable or unwilling to publish the extension to link:https://open-vsx.org[open-vsx.org], and if there is no Open VSX equivalent of the extension, consider link:https://github.com/open-vsx/publish-extensions/issues[reporting an issue] to the Open VSX team.
+
+* If you have a closed-source extension or an extension developed only for internal use in your company, you can add the extension directly from a `.vsix` file that is accessible from a URL while building the plugin registry container:
```
```suggestion
* If you have a closed-source extension or an extension developed only for internal use in your company, you can add the extension directly from a `.vsix`file by using a URL accessible to your custom plugin registry container build:
```

max-cx replied:
@nickboldt, I took the liberty to edit your suggestion ^. Is this okay this way?

nickboldt replied:
well, the "build" at the end is redundant but valid, so +1.

nickboldt commented on modules/administration-guide/partials/proc_selecting-an-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,29 @@
+[id="selecting-an-open-vsx-registry-instance"]
+= Selecting an Open VSX registry instance
+
+The Open VSX registry at `pass:c,a,q[https://open-vsx.org]` is the default if it is resolved from within your cluster. If not, then the embedded Open VSX registry within the {prod-short} `plugin-registry` pod is the default.
```
```suggestion
The Open VSX registry at `pass:c,a,q[https://open-vsx.org]` is the default if resolved from within your cluster. If not, then the embedded Open VSX registry within the {prod-short} `plugin-registry` pod is the default.
```

nickboldt commented on modules/administration-guide/partials/proc_selecting-an-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,29 @@
+ 
+* To select the embedded Open VSX registry instance in the `plugin-registry` pod, use `openVSXURL: ''`. See the next section for how to customize the included extensions.
+
+* You can point `openVSXURL` at the URL of a standalone Open VSX registry instance. 
```
```suggestion
* You can also point `openVSXURL` at the URL of a standalone Open VSX registry instance, if its URL is accessible from within your cluster and not blocked by a proxy. 
```

max-cx replied:
@nickboldt, can I replace `provided` with `if`? `[Yes/No]`

max-cx committed:
Update modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

Co-authored-by: Nick Boldt <nboldt@redhat.com>

Update modules/administration-guide/partials/proc_selecting-an-open-vsx-registry-instance.adoc

Co-authored-by: Nick Boldt <nboldt@redhat.com>

Update modules/administration-guide/partials/proc_selecting-an-open-vsx-registry-instance.adoc

Update modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

max-cx commented on modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,87 @@
+====
+* If the extension publisher is unavailable or unwilling to publish the extension to link:https://open-vsx.org[open-vsx.org], and if there is no Open VSX equivalent of the extension, consider link:https://github.com/open-vsx/publish-extensions/issues[reporting an issue] to the Open VSX team.
+
+* If you have a closed-source extension or an extension developed only for internal use in your company, you can add the extension directly from a `.vsix`file by using a URL accessible to your custom plugin registry container build:
```
```suggestion
* If you have a closed-source extension or an extension developed only for internal use in your company, you can add the extension directly from a `.vsix`file by using a URL accessible to your custom plugin registry container:
```

max-cx committed:
Update modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

max-cx commented on modules/administration-guide/partials/proc_selecting-an-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,29 @@
+ 
+* To select the embedded Open VSX registry instance in the `plugin-registry` pod, use `openVSXURL: ''`. See the next section for how to customize the included extensions.
+
+* You can also point `openVSXURL` at the URL of a standalone Open VSX registry instance, if its URL is accessible from within your cluster and not blocked by a proxy. 
```
```suggestion
* You can also point `openVSXURL` at the URL of a standalone Open VSX registry instance if its URL is accessible from within your cluster and not blocked by a proxy. 
```

max-cx committed:
Update modules/administration-guide/partials/proc_selecting-an-open-vsx-registry-instance.adoc

max-cx commented on modules/administration-guide/pages/extensions-for-microsoft-visual-studio-code-open-source.adoc:
```diff
@@ -0,0 +1,18 @@
+[id="extensions-for-microsoft-visual-studio-code-open-source"]
+= Extensions for Microsoft Visual Studio Code - Open Source
+
+To manage extensions, the link:https://github.com/microsoft/vscode[Microsoft Visual Studio Code - Open Source IDE] cloud editors use one of these link:https://open-vsx.org/about[Open VSX] registries:
```
```suggestion
To manage extensions, the IDE that loads in workspaces uses an link:https://open-vsx.org/about[Open VSX] registry instance:
```

nickboldt replied:
why "a" ? there's only one code-rhel8 image per release per deployment.

nickboldt replied:
"to manage exetensions, the editor uses an openvsx registry" is a lot cleaner IMHO.

nickboldt replied:
"IDE that loads in workspaces" == "editor" :D 

why so wordy?

https://imgflip.com/i/77vsq9 :D

nickboldt replied:
```suggestion
To manage extensions, the editor uses an link:https://open-vsx.org/about[Open VSX] registry:
```

max-cx replied:
:) Yes, but that's not typical for editors to load in workspaces. Combine that with the fact that users only visit the fewest docs page they can only when they really need something, I can't expect user guide readers to come from the same place as a tech lead on this project (of course they can learn it somewhere, which happens to be the docs, which is this file).

nickboldt replied:
> not typical for editors to load in workspaces

ok now I'm confused. 

```
if 
   IDE = editor,
     and 
   workspace = filesystem w/ an editor (including extensions) and 1 or more checked out projects, 
then
  surely every workspace has an editor;
else 
  what would you use to edit files in the workspace? the terminal?
```

Do we have a "no editor" option when loading a workspace?

max-cx replied:
On second thought, maybe I should add the VS Code name back into my suggestion because otherwise it's reads to general.

max-cx replied:
```suggestion
To manage extensions, the link:https://github.com/microsoft/vscode[Microsoft Visual Studio Code - Open Source] IDE of the workspace uses an link:https://open-vsx.org/about[Open VSX] registry instance:
```

max-cx commented on modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,87 @@
+
+. Check that the `plugin-registry` pod has restarted and is running.
+
+. Restart the workspace and check the available extensions in the *Extensions* view of the Microsoft Visual Studio Code - Open Source in the workspace.
```
```suggestion
. Restart the workspace and check the available extensions in the *Extensions* view of the workspace IDE.
```

max-cx committed:
Update modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

max-cx committed:
Update modules/administration-guide/pages/extensions-for-microsoft-visual-studio-code-open-source.adoc

max-cx commented on modules/administration-guide/pages/extensions-for-microsoft-visual-studio-code-open-source.adoc:
```diff
@@ -0,0 +1,18 @@
+[id="extensions-for-microsoft-visual-studio-code-open-source"]
+= Extensions for Microsoft Visual Studio Code - Open Source
+
+To manage extensions, the link:https://github.com/microsoft/vscode[Microsoft Visual Studio Code - Open Source] IDE of the workspace uses an link:https://open-vsx.org/about[Open VSX] registry instance:
```
```suggestion
To manage extensions, link:https://github.com/microsoft/vscode[this IDE] uses one of these link:https://open-vsx.org/about[Open VSX] registry instances:
```

max-cx committed:
Update modules/administration-guide/pages/extensions-for-microsoft-visual-studio-code-open-source.adoc

max-cx commented on modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,87 @@
+====
+* If the extension publisher is unavailable or unwilling to publish the extension to link:https://open-vsx.org[open-vsx.org], and if there is no Open VSX equivalent of the extension, consider link:https://github.com/open-vsx/publish-extensions/issues[reporting an issue] to the Open VSX team.
+
+* If you have a closed-source extension or an extension developed only for internal use in your company, you can add the extension directly from a `.vsix`file by using a URL accessible to your custom plugin registry container:
```
```suggestion
* If you have a closed-source extension or an extension developed only for internal use in your company, you can add the extension directly from a `.vsix` file by using a URL accessible to your custom plugin registry container:
```

max-cx committed:
Update modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

max-cx commented on modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,87 @@
+----
+====
++
+NOTE: Please read the link:https://aka.ms/vsmarketplace-ToU[Terms of Use] for the link:https://marketplace.visualstudio.com/VSCode[Microsoft Visual Studio Marketplace] before using its resources.
```
```suggestion
WARNING: Please read the link:https://aka.ms/vsmarketplace-ToU[Terms of Use] for the link:https://marketplace.visualstudio.com/VSCode[Microsoft Visual Studio Marketplace] before using its resources.
```

max-cx committed:
Update modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

max-cx commented on modules/administration-guide/partials/proc_selecting-an-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,29 @@
+
+The Open VSX registry at `pass:c,a,q[https://open-vsx.org]` is the default if resolved from within your cluster. If not, then the embedded Open VSX registry within the {prod-short} `plugin-registry` pod is the default.
+
+If the default Open VSX registry instance is not what you need, you can select another Open VSX registry instance:
```
```suggestion
If the default Open VSX registry instance is not what you need, you can select another Open VSX registry instance as follows.
```

max-cx committed:
Update modules/administration-guide/partials/proc_selecting-an-open-vsx-registry-instance.adoc

max-cx commented on modules/administration-guide/partials/proc_selecting-an-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,29 @@
+[id="selecting-an-open-vsx-registry-instance"]
+= Selecting an Open VSX registry instance
+
+The Open VSX registry at `pass:c,a,q[https://open-vsx.org]` is the default if resolved from within your cluster. If not, then the embedded Open VSX registry within the {prod-short} `plugin-registry` pod is the default.
```
```suggestion
The Open VSX registry at `pass:c,a,q[https://open-vsx.org]` is the default if resolved from within your organization's cluster. If not, then the embedded Open VSX registry within the {prod-short} `plugin-registry` pod is the default.
```

max-cx commented on modules/administration-guide/partials/proc_selecting-an-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,29 @@
+ 
+* To select the embedded Open VSX registry instance in the `plugin-registry` pod, use `openVSXURL: ''`. See the next section for how to customize the included extensions.
+
+* You can also point `openVSXURL` at the URL of a standalone Open VSX registry instance if its URL is accessible from within your cluster and not blocked by a proxy. 
```
```suggestion
* You can also point `openVSXURL` at the URL of a standalone Open VSX registry instance if its URL is accessible from within your organization's cluster and not blocked by a proxy. 
```

max-cx committed:
Apply suggestions from code review

max-cx commented on modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,87 @@
+$ {docker-cli} push quay.io/__<username/plugin_registry:custom>__
+----
+
+. Edit the `CheCluster` custom resource in the cluster to point to the image (for example, on link:https://quay.io/[quay.io]) and then save the changes:
```
```suggestion
. Edit the `CheCluster` custom resource in your organization's cluster to point to the image (for example, on link:https://quay.io/[quay.io]) and then save the changes:
```

max-cx committed:
Update modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

max-cx commented on modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,87 @@
+:_content-type: PROCEDURE
+
+[id="adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance"]
+= Adding or removing pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance
```
```suggestion
= Adding or removing extensions in the embedded Open VSX registry instance
```

max-cx commented on modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,87 @@
+:_content-type: PROCEDURE
+
+[id="adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance"]
```
```suggestion
[id="adding-or-removing-extensions-in-the-embedded-open-vsx-registry-instance"]
```

max-cx committed:
Apply suggestions from code review

max-cx commented on modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc:
```diff
@@ -0,0 +1,87 @@
+
+include::example$snip_{project-context}-steps-to-add-or-remove-extensions-in-openvsx-sync-json.adoc[]
+
+** If the extension is published on link:https://open-vsx.org[open-vsx.org], you can add its extension `id` in the format __<published_by>__.__<unique_identifier>__. You can find the `id` on the extension's listing page on link:https://open-vsx.org[open-vsx.org].
```
```suggestion
** If the extension is published on link:https://open-vsx.org[open-vsx.org], you can add its extension `id` in the format __<published_by>__.__<unique_identifier>__. You can find the `id` details on the extension's listing page on link:https://open-vsx.org[open-vsx.org].
```

max-cx committed:
Apply suggestions from code review

max-cx committed:
edit
